includes/abstracts/llms-abstract-generator-posts.php

```diff
@@ -542,35 +528,79 @@ protected function handle_reusable_blocks( $post, $raw ) {
 		// Importing blocks is disabled.
 		if ( ! $this->is_reusable_block_importing_enabled() ) {
 			return null;
+		} elseif ( empty( $raw['_extras']['blocks'] ) ) {
+			return false;
 		}
 
-		if ( ! empty( $raw['_extras']['blocks'] ) ) {
-
-			$find    = array();
-			$replace = array();
-			$post_id = $post->get( 'id' );
-			foreach ( $raw['_extras']['blocks'] as $block_id => $block ) {
+		$find    = array();
+		$replace = array();
+		foreach ( $raw['_extras']['blocks'] as $block_id => $block ) {
 
-				$new_id = $this->create_reusable_block( $block_id, $block );
-				if ( ! is_wp_error( $new_id ) && is_numeric( $new_id ) ) {
-					$find[]    = sprintf( '<!-- wp:block {"ref":%d}', absint( $block_id ) );
-					$replace[] = sprintf( '<!-- wp:block {"ref":%d}', $new_id );
-				}
+			$new_id = $this->create_reusable_block( $block_id, $block );
+			if ( ! is_wp_error( $new_id ) && is_numeric( $new_id ) ) {
+				$find[]    = sprintf( '<!-- wp:block {"ref":%d}', absint( $block_id ) );
+				$replace[] = sprintf( '<!-- wp:block {"ref":%d}', $new_id );
 			}
+		}
 
-			if ( $find && $replace ) {
-				$args = array(
-					'ID'           => $post->get( 'id' ),
-					'post_content' => str_replace( $find, $replace, $post->get( 'content', true ) ),
-				);
-				return wp_update_post( $args ) ? true : false;
-			}
+		if ( $find && $replace ) {
+			$args = array(
+				'ID'           => $post->get( 'id' ),
+				'post_content' => str_replace( $find, $replace, $post->get( 'content', true ) ),
+			);
+			return wp_update_post( $args ) ? true : false;
 		}
 
 		return false;
 
 	}
 
+	/**
+	 * Insert a reusable block into the database
+	 *
+	 * @since [version]
+	 *
+	 * @param int   $block_id WP_Post ID of the block being imported. This will be the ID as found on the original site.
+	 * @param array $block    {
+	 *     Array of block data.
+	 *
+	 *     @type string $title   Title of the reusable block.
+	 *     @type string $content Content of the reusable block.
+	 * }
+	 * @return WP_Error|int An error object if an error is encountered creating the block or
+	 *                      the new block's WP_Post ID as an integer on success.
+	 */
+	protected function insert_resuable_block( $block_id, $block ) {
+
+		$id = wp_insert_post(
+			array(
+				'post_content' => $block['content'],
+				'post_title'   => $block['title'],
+				'post_type'    => 'wp_block',
+				'post_status'  => 'publish',
+			)
+		);
+
+		if ( ! is_wp_error( $id ) ) {
```

Comment:
This only makes sense if we insert the post asking for returning a wp error on failure. 
Meaning that the previous instruction must be:
```php
		$id = wp_insert_post(
			array(
				'post_content' => $block['content'],
				'post_title'   => $block['title'],
				'post_type'    => 'wp_block',
				'post_status'  => 'publish',
			),
			true
		);
```
otherwise, this must be just
```suggestion
		if ( $id ) {
```

Reply:
mmm silly me. Thanks